src/sentry/utils/snuba.py

```diff
@@ -1239,7 +1256,38 @@ def _bulk_snuba_query(snuba_requests: Sequence[SnubaRequest]) -> ResultSet:
                 if body.get("error"):
                     error = body["error"]
                     if response.status == 429:
+                        if options.get("issues.use-snuba-error-data"):
+                            if "stats" not in error:
+                                # Should not hit this - snuba always returns stats when there is an error
+                                raise RateLimitExceeded(error["message"])
+
+                            quota_allowance_summary = error["stats"]["quota_allowance"]["details"][
+                                "summary"
+                            ]
+                            rejected_by = quota_allowance_summary["rejected_by"]
+                            throttled_by = quota_allowance_summary["throttled_by"]
+
+                            if rejected_by:
+                                raise RateLimitExceeded(
+                                    error["message"],
+                                    policy=rejected_by["policy"],
+                                    quota_unit=rejected_by["quota_unit"],
+                                    storage_key=rejected_by["storage_key"],
+                                    quota_used=rejected_by["quota_used"],
+                                    rejection_threshold=rejected_by["rejection_threshold"],
+                                )
+                            elif throttled_by:
+                                raise RateLimitExceeded(
+                                    error["message"],
+                                    policy=throttled_by["policy"],
+                                    quota_unit=throttled_by["quota_unit"],
+                                    storage_key=throttled_by["storage_key"],
+                                    quota_used=throttled_by["quota_used"],
+                                    rejection_threshold=throttled_by["rejection_threshold"],
+                                )
```

Comment:
this kind of code that's needed for informational purposes, you want to wrap in a try catch so it always fails open. As I said before, this is a critical path piece of code, you can fail every query if something goes wrong here. Your code is a nice to have, not a critical feature